include/uxr/agent/Agent.hpp

```diff
@@ -0,0 +1,341 @@
+        INCOMPATIBLE_ERROR      = 0x86,
+        /** Indicates a failure in the operation due to a resource error on the Agent. */
+        RESOURCES_ERROR         = 0x87
+    };
+
+    /**
+     * @brief These flags determines the behaviour of the create operation.
+     *        According to the DDS-XRCE standart, the following modes are defined:
+     *
+     *        | REUSE_MODE | REPLACE_MODE | Exists |  Behaviour                                                  |
+     *        |:----------:|:------------:|:------:|:------------------------------------------------------------|
+     *        | no matter  |   no matter  |   no   | The XRCEObject is created normally.                         |
+     *        |   false    |     false    |   yes  | Nothing is done and an ALREADY_EXISTS_ERROR is returned.    |
+     *        |   false    |     true     |   yes  | The existing XRCEObject is remove and a new one is creates. |
+     *        |   true     |     false    |   yes  | Nothing is done and an OK_MATCHED is returned in case of matching, and a MISMATCH_ERROR in other case. |
+     *        |   true     |     true     |   yes  | Nothing is done and an OK_MATCHED is returned in case of matching, and a new XRCEObject and an OK is returned in other case. |
+     */
+    enum CreationFlag : uint8_t
+    {
+        REUSE_MODE      = 0x01 << 1,
+        REPLACE_MODE    = 0x01 << 2
+    };
+
+    /**
+     * @brief The identifiers of the XRCEObject types.
+     */
+    enum ObjectKind : uint8_t
+    {
+        /** Identifies a Participant. */
+        PARTICIPANT_OBJK    = 0x01,
+        /** Identifies a Topic. */
+        TOPIC_OBJK          = 0x02,
+        /** Identifies a Publisher. */
+        PUBLISHER_OBJK      = 0x03,
+        /** Identifies a Subscriber. */
+        SUBSCRIBER_OBJK     = 0x04,
+        /** Identifies a DataWriter. */
+        DATAWRITER_OBJK     = 0x05,
+        /** Identifies a DataReader. */
+        DATAREADER_OBJK     = 0x06
+    };
+
+    /**
+     * @brief Generates an XRCEObject identifier from a prefix and an ObjectKind identifier.
+     * @param prefix        The XRCEObject's prefix.
+     * @param object_kind   The XRCEObject's kind.
+     * @return The XRCEObject's identifier.
+     */
+    static uint16_t get_object_id(uint16_t prefix, ObjectKind object_kind)
+    {
+        return uint16_t((uint16_t(prefix) << 4) + object_kind);
+    }
+
+    Agent() = delete;
+    ~Agent() = delete;
+
+    /**
+     * @brief Creates a ProxyClient which can be reused by an external Client.
+     * @param key       The identifier of the ProxyClient.
+     * @param session   The identifier of the Session attached to the ProxyClient.
+     * @param mtu       The Maximum Transmission Unit (MTU) of the Session.
+     * @param errcode   The result status of the operation.
+     * @return  true in case of success and false in other case.
+     */
+    microxrcedds_agent_DllAPI static bool create_client(
```

Comment:
I think that this define can be renamed to something shorter. It does not provide any information to the API user reader and it takes a lot of space. Also, I think that if it is a define, it should use capital letters.

Reply:
Yeah, great idea!